<!-- Move Base launch file -->

<launch>
    <arg name="local" default="true"/>

    <machine name="movo1" address="movo1" user="movo" 
                 env-loader="/home/movo/env.sh" default="true" unless="$(arg local)"/>
                 
    <arg name="no_static_map" default="true"/>

    <arg name="global_frame" default="map" />
    <arg name="eband_rotational_threshold_multiplier" default="1.0" />
    <arg name="eband_disallow_hysteresis" default="false" />
    <arg name="map_topic" default="/map" />

    <node pkg="move_base" type="move_base" name="move_base" clear_params="true">
    <!-- Remap into correct namespace for mux -->
        <remap from="cmd_vel" to="/movo/navigation/cmd_vel" />
        
        <!-- Change the name of the move_base; it is first checked and handled by the client that we're using -->
		<remap from="move_base/goal" to="move_base_navi/goal" />
		<remap from="move_base/result" to="move_base_navi/result" />
		<remap from="move_base/feedback" to="move_base_navi/feedback" />
		<remap from="move_base/status" to="move_base_navi/status" />
		<remap from="move_base/cancel" to="move_base_navi/cancel" />
		<remap from="move_base_simple/goal" to="move_base_navi_simple/goal"/>
		
        <!-- Remap the odometry feedback topic -->
        <remap from="/odom" to="/movo/odometry/local_filtered" />

        <!-- Load common configuration files -->
        <rosparam file="$(find movo_demos)/config/2d_nav/eband_planner_params.yaml" command="load" />
        <rosparam file="$(find movo_demos)/config/2d_nav/global_planner_params.yaml" command="load" />
        <rosparam file="$(find movo_demos)/config/2d_nav/move_base_params.yaml" command="load" />
        <rosparam file="$(find movo_demos)/config/2d_nav/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find movo_demos)/config/2d_nav/costmap_common_params.yaml" command="load" ns="local_costmap" />

        <!-- Load global/local costmap specific parameters -->
        <rosparam file="$(find movo_demos)/config/2d_nav/costmap_local.yaml" command="load"/>
        <rosparam file="$(find movo_demos)/config/2d_nav/costmap_global_sensors.yaml" command="load" if="$(arg no_static_map)"/>
        <rosparam file="$(find movo_demos)/config/2d_nav/costmap_global_map.yaml" command="load" unless="$(arg no_static_map)"/>

        <param name="base_local_planner" value="eband_local_planner/EBandPlannerROS" />
        <param name="base_global_planner" value="global_planner/GlobalPlanner" />
        
        <param name="global_costmap/map_topic" value="$(arg map_topic)" unless="$(arg no_static_map)" />
        
        <param name="local_costmap/global_frame" value="map" unless="$(arg no_static_map)" />
        <param name="global_costmap/global_frame" value="map" unless="$(arg no_static_map)" />
        <param name="local_costmap/global_frame" value="odom" if="$(arg no_static_map)" />
        <param name="global_costmap/global_frame" value="odom" if="$(arg no_static_map)" />

        <param name="EBandPlannerROS/rotational_threshold_multiplier" value="$(arg eband_rotational_threshold_multiplier)" />
        <param name="EBandPlannerROS/disallow_hysteresis" value="$(arg eband_disallow_hysteresis)" />
    </node>
    
</launch> 
           


